spharpy/samplings/coordinates.py

```
@@ -310,3 +314,29 @@ def n_max(self, value):
             self._n_max = None
         else:
             self._n_max = int(value)
+
+    def _set_weights(self, weights):
+        """Check and set sampling weights.
+
+        Set self._weights, which is an atleast_1d numpy array of shape
+        [L,M,...,N].
+
+        """
+        # check if weights is None
+        if weights is None:
+            self._weights = weights
+            return
+
+        weights = np.asarray(weights, dtype=np.float64)
+
+        # reshape according to self._points
+        if weights.size != self.csize:
+            raise ValueError("Weights must have same size as csize.")
+        weights = weights.reshape(self.cshape)
+
+        if not np.isclose(np.sum(weights), 4*np.pi, atol=1e-6, rtol=1e-6):
```

You don’t check for negativity or nans. Unnormalized or negative weights could silently pass the sum‐check yet break algorithms downstream.

---

Good point!